### PR TITLE
Add chromosome and region query for g3d format

### PR DESCRIPTION
**Pull request by lidaof**
add chromosome and region query for g3d format with examples below.

show struct from chr7:
```js
components.apply(StateTransforms.Model.StructureSelectionFromExpression, { expression: g3dChromosomeQuery('chr7'), label: 'chr7' })
        .apply(StateTransforms.Representation.StructureRepresentation3D, reprChrom);
```
show struct from chr7:2000000-7000000:
```js
    components.apply(StateTransforms.Model.StructureSelectionFromExpression, { expression: g3dRegionQuery('chr7', 2000000, 7000000), label: 'chr7:2000000-7000000' })
        .apply(StateTransforms.Representation.StructureRepresentation3D, reprRegion);
```

### CODE REVIEW

File: src/extensions/g3d/model.ts [hunk 1]
```diff
@@ -174,6 +174,23 @@ export const G3dSymbols = {
             const seqId = ctx.element.unit.model.coarseHierarchy.spheres.seq_id_begin.value(ctx.element.element);
             return info.haplotype[seqId] || '';
         }
+    ),
+    chromosome: QuerySymbolRuntime.Dynamic(CustomPropSymbol('g3d', 'chromosome', Type.Str),
```
Comment (arose) [inline src/extensions/g3d/model.ts:178]: @dsehnal, looks like we don't expose the coarse properties, probably should

File: src/extensions/g3d/model.ts [hunk 2]
```diff
@@ -174,6 +174,23 @@ export const G3dSymbols = {
...
+        ctx => {
+            if (Unit.isAtomic(ctx.element.unit)) return '';
+            const {  asym_id } = ctx.element.unit.model.coarseHierarchy.spheres;
+            return asym_id.value(ctx.element.element) || '';
+        }
+    ),
+    region: QuerySymbolRuntime.Dynamic(CustomPropSymbol('g3d', 'region', Type.Str),
+        ctx => {
+            if (Unit.isAtomic(ctx.element.unit)) return '';
```
Comment (arose) [inline src/extensions/g3d/model.ts:187]: Should this function always return a number?

File: src/extensions/g3d/model.ts [hunk 3]
```diff
@@ -185,6 +202,19 @@ export function g3dHaplotypeQuery(haplotype: string) {
...
 
+export function g3dChromosomeQuery(chromosome: string) {
+    return MS.struct.generator.atomGroups({
+        'chain-test': MS.core.rel.eq([G3dSymbols.chromosome.symbol(), chromosome]),
+    });
+}
+
+export function g3dRegionQuery(chr: string, start: number, end: number) {
+    return MS.struct.generator.atomGroups({
+        'chain-test': MS.core.rel.eq([G3dSymbols.chromosome.symbol(), chr]),
```
Comment (arose) [inline src/extensions/g3d/model.ts:213]: You can filter for sphere units in the chain test so the residue test would not be called needlessly (same for g3dChromosomeQuery to catch the edge case of an empty chain/chromosome name better). 

```ts
'chain-test': MS.core.logic.and([
    MS.core.rel.eq([MS.ammp('objectPrimitive'), 'sphere']),
    MS.core.rel.eq([G3dSymbols.chromosome.symbol(), chr])
])
```

File: src/extensions/g3d/model.ts (hunk 1 above)
Comment (dsehnal) [reply src/extensions/g3d/model.ts:178]: Yeah, we should. Not in scope of this PR tho.

File: src/extensions/g3d/model.ts (hunk 3 above)
Comment (lidaof) [reply src/extensions/g3d/model.ts:213]: Thank you for comments @arose.  can you please give an example or modify the PR? I am not familiar with the mol* codebase, this was got from @dsehnal's example code before. good it's working :) Thank both of you!

Comment (arose) [reply src/extensions/g3d/model.ts:213]: Well, just swap out the two 'chain-test's with the one I gave above and try to see if it still does what you want

Comment (lidaof) [reply src/extensions/g3d/model.ts]: I see, thanks @arose, yes, it still works as what I want!

File: src/extensions/g3d/model.ts [hunk 4]
```diff
@@ -174,6 +174,23 @@ export const G3dSymbols = {
...
         }
+    ),
+    chromosome: QuerySymbolRuntime.Dynamic(CustomPropSymbol('g3d', 'chromosome', Type.Str),
+        ctx => {
+            if (Unit.isAtomic(ctx.element.unit)) return '';
+            const { asym_id } = ctx.element.unit.model.coarseHierarchy.spheres;
+            return asym_id.value(ctx.element.element) || '';
+        }
+    ),
+    region: QuerySymbolRuntime.Dynamic(CustomPropSymbol('g3d', 'region', Type.Str),
```
Comment (arose) [inline src/extensions/g3d/model.ts:185]: I think this should be `Type.Num` not `Type.Str` to match the type of info.start[seqId]

File: src/extensions/g3d/model.ts (hunk 2 above)
Comment (arose) [reply src/extensions/g3d/model.ts:187]: this should return a number or throw

File: src/extensions/g3d/model.ts [hunk 5]
```diff
@@ -174,6 +174,23 @@ export const G3dSymbols = {
...
+        ctx => {
+            if (Unit.isAtomic(ctx.element.unit)) return '';
+            const info = (G3dInfoDataProperty as any).get(ctx.element.unit.model);
+            if (!info) return '';
```
Comment (arose) [inline src/extensions/g3d/model.ts:189]: should return a number or throw

File: src/extensions/g3d/model.ts [hunk 6]
```diff
@@ -174,6 +174,23 @@ export const G3dSymbols = {
...
+            const seqId = ctx.element.unit.model.coarseHierarchy.spheres.seq_id_begin.value(ctx.element.element);
+            const s = info.start[seqId];
```
Comment (arose) [inline src/extensions/g3d/model.ts:191]: maybe just `return info.start[seqId] || 0;`

Comment (lidaof) [reply src/extensions/g3d/model.ts]: thank you @arose for these nice comments!

**Comment by lidaof** [comment]
trying added chromosome array to the G3dInfoData object, as I need the chr and start and resolution for data query to bigwig and bigbed files for decoration. please review, thank you so much!

**Comment by lidaof** [comment]
Hi Alex and David,

do you think this PR could be merged? Thank you so much!

**Comment by lidaof** [comment]
Thank you!!
